src/MongoDB.Driver.Core/Core/Operations/EndTransactionOperation.cs

```diff
@@ -64,6 +76,9 @@ public MessageEncoderSettings MessageEncoderSettings
         /// </value>
         public WriteConcern WriteConcern => _writeConcern;
 
+        // internal properties
```

Comment:
Minor comment:
Not sure about ordering whether `internal` or `protected` properties should be placed first? But in ` public abstract class MongoCollection` class, `protected` constructors are placed before `internal`. So, it's up to you.

Reply:
Hm, good question. At a quick glance, I couldn't find any guidance from Microsoft but [StackOverflow](https://stackoverflow.com/questions/150479/order-of-items-in-classes-fields-properties-constructors-methods) links to this [StyleCop suggestion](http://stylecop.soyuz5.com/SA1202.html) that `internal` should come before `protected`.

Reply:
Ok, maybe we can use this ordering.

Reply:
My recommended ordering is:

1. public
2. protected (it's public to its subclasses)
3. internal (I would prefer that there not be any of these)
4. private

These are in decreasing order of visibility.